### DISCUSSION TITLE
Casting byte array to Float32

### DISCUSSION THREAD

**Discussion by jblindsay** [modular/modular#2624] [Q&A] [answered]
Hello, I'm wondering how I can go about casting an array of four bytes into a Float32 in Mojo? Essentially, I'm trying to replicate some old Python code that looked like this:

```python
val, = struct.unpack(pack_format, data[pos:pos + data_size])
```

**Comment by ericqu** [comment, marked as answer]
I am not sure if that is the intended way to do that; I am just trying things out. I used `bitcast` and reused the `SIMD` type to force a cast. To illustrate, I start from a Float32 to an array of uint8 (byte) and from that array back to a Float32.

```
from memory.unsafe import bitcast

fn main () -> None :
    var temp_array: SIMD[DType.uint8, 4] = bitcast[DType.uint8, 4](SIMD[DType.float32, 1](3.1))
    print(temp_array)
    var test_float32: Float32 = bitcast[DType.float32, 1](temp_array)
    print(test_float32)
```

**Reply by jblindsay** [reply]
That's helpful, thanks!
